google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -93,12 +97,302 @@ public BucketInfo apply(com.google.api.services.storage.model.Bucket pb) {
+                      : transform(
+                          condition.getMatchesStorageClass(),
+                          new Function<String, StorageClass>() {
+                            public StorageClass apply(String storageClass) {
+                              return StorageClass.valueOf(storageClass);
+                            }
+                          }));
+
+      return new LifecycleRule(lifecycleAction, conditionBuilder.build());
+    }
+
+    /**
+     * Condition for a Lifecycle rule, specifies under what criteria an Action should be executed
+     *
+     * @see <a href="https://cloud.google.com/storage/docs/lifecycle">Object Lifecycle Management</a>
+     */
+    public static class LifecycleCondition implements Serializable {
+      private static final long serialVersionUID = -6482314338394768785L;
+      private final Integer age;
+      private final DateTime createdBefore;
+      private final Integer numberOfNewerVersions;
+      private final Boolean isLive;
+      private final List<StorageClass> matchesStorageClass;
+
+      private LifecycleCondition(Builder builder) {
+        this.age = builder.age;
+        this.createdBefore = builder.createdBefore;
+        this.numberOfNewerVersions = builder.numberOfNewerVersions;
+        this.isLive = builder.isLive;
+        this.matchesStorageClass = builder.matchesStorageClass;
+      }
+
+      public Builder toBuilder() {
+        return newBuilder()
+                .setAge(this.age)
+                .setCreatedBefore(this.createdBefore)
+                .setNumberOfNewerVersions(this.numberOfNewerVersions)
+                .setIsLive(this.isLive)
+                .setMatchesStorageClass(this.matchesStorageClass);
+      }
+
+      public static Builder newBuilder() {
+        return new Builder();
+      }
+
+      public Integer getAge() {
+        return age;
+      }
+
+      public DateTime getCreatedBefore() {
+        return createdBefore;
+      }
+
+      public Integer getNumberOfNewerVersions() {
+        return numberOfNewerVersions;
+      }
+
+      public Boolean getIsLive() {
+        return isLive;
+      }
+
+      public List<StorageClass> getMatchesStorageClass() {
+        return matchesStorageClass;
+      }
+
+      /** Builder for {@code LifecycleCondition}. */
+      public static class Builder {
+        private Integer age;
+        private DateTime createdBefore;
+        private Integer numberOfNewerVersions;
+        private Boolean isLive;
+        private List<StorageClass> matchesStorageClass;
+
+        private Builder() {}
+
+        /** Sets the age in days a Blob should be for an Action to be executed **/
+        public Builder setAge(Integer age) {
+          this.age = age;
+          return this;
+        }
+
+        /** Sets the date a Blob should be created before for an Action to be executed. Note that only the date
+         * will be considered, if the time is specified it will be truncated **/
```

Comment:
Append `This condition is satisfied when an object is created before midnight of the specified date in UTC.`